### DISCUSSION TITLE
Unintended focus control occurs in NavigationPage.

### DISCUSSION THREAD

**Discussion by cat0363** [dotnet/maui#14210] [Q&A]
If NavigationPage is specified for MainPage, unintended focus control will occur.
If Shell is specified for MainPage, unintended focus control will not occur.
To verify this problem, I created the screen layout shown below.

[MainPage.xaml]

    <ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="NavigationPageFocusTest.MainPage">
        <Grid RowDefinitions="44,44,150,*" RowSpacing="5">
            <StackLayout Grid.Row="0" Orientation="Horizontal">
                <Label Text="Role:" VerticalOptions="Center" />
                <Picker VerticalOptions="Center">
                    <Picker.ItemsSource>
                        <x:Array Type="{x:Type x:String}">
                            <x:String>President</x:String>
                            <x:String>Executive Vice President</x:String>
                            <x:String>Executive Managing Director</x:String>
                            <x:String>General Manager</x:String>
                            <x:String>Manager</x:String>
                            <x:String>Section Chief</x:String>
                            <x:String>Deputy Section Manager</x:String>
                            <x:String>Employee</x:String>
                        </x:Array>
                    </Picker.ItemsSource>
                </Picker>
            </StackLayout>
            <Button Grid.Row="1" x:Name="btnAdd" Text="Add Employee" BackgroundColor="Blue" TextColor="White" Clicked="btnAdd_Clicked"/>
            <ScrollView Grid.Row="2" x:Name="svEmployeeList" Orientation="Vertical">
                <StackLayout x:Name="slEmployeeList" Orientation="Vertical" BindableLayout.ItemsSource="{Binding Employees}">
                    <BindableLayout.ItemTemplate>
                        <DataTemplate>
                            <Grid ColumnDefinitions="80,*" HeightRequest="44">
                                <Label Grid.Column="0" Text="{Binding EmployeeId}" VerticalOptions="Center" />
                                <Entry Grid.Column="1" Text="{Binding EmployeeName}" VerticalOptions="Center" />
                            </Grid>
                        </DataTemplate>
                    </BindableLayout.ItemTemplate>
                </StackLayout>
            </ScrollView>
        </Grid>    
    </ContentPage>

It is a simple screen layout with Picker, Button, Label and Entry in list format.
Pressing the Add Employee button adds a new employee to the employee list and 
scrolls to the bottom of the employee list.

MainPage's internal logic and various classes are shown below.

[MainPage.xaml.cs]

    using System.ComponentModel;
    using System.Runtime.CompilerServices;
    namespace NavigationPageFocusTest;

    public partial class MainPage : ContentPage
    {
        /// <summary>
        /// Employee
        /// </summary>
        public class Employee : INotifyPropertyChanged
        {
            /// <summary>
            /// Property Changed Event
            /// </summary>
            public event PropertyChangedEventHandler PropertyChanged;
            /// <summary>
            /// Property Changed 
            /// </summary>
            /// <param name="name">Property Name</param>
            public void OnPropertyChanged([CallerMemberName] string name = "") => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));

            /// <summary>
            /// Employee Id
            /// </summary>
            private int pEmployeeId;
            /// <summary>
            /// Employee Name
            /// </summary>
            private string pEmployeeName;

            /// <summary>
            /// Employee Id
            /// </summary>
            public int EmployeeId
            {
                get 
                { 
                    return pEmployeeId; 
                }
                set 
                { 
                    if (pEmployeeId != value)
                    {
                        pEmployeeId = value;
                        OnPropertyChanged(nameof(EmployeeId));
                    }
                }
            }
            /// <summary>
            /// Employee Name
            /// </summary>
            public string EmployeeName
            {
                get
                {
                    return pEmployeeName;
                }
                set
                {
                    if (pEmployeeName != value)
                    {
                        pEmployeeName = value;
                        OnPropertyChanged(nameof(EmployeeName));
                    }
                }
            }
        }

        /// <summary>
        /// MainPage View Model
        /// </summary>
        public class ViewModelMainPage : INotifyPropertyChanged
        {
            /// <summary>
            /// Property Changed Event
            /// </summary>
            public event PropertyChangedEventHandler PropertyChanged;
            /// <summary>
            /// Property Changed 
            /// </summary>
            /// <param name="name">Property Name</param>
            public void OnPropertyChanged([CallerMemberName] string name = "") =>
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));

            /// <summary>
            /// Employee List
            /// </summary>
            private List<Employee> pEmployeeList;
        
            /// <summary>
            /// Employee List (Orderd EmployeeId)
            /// </summary>
            public IEnumerable<Employee> Employees
            {
                get
                {
                    return pEmployeeList.AsEnumerable()
                        .OrderBy(
                            x => x.EmployeeId
                        );
                }
            }

            /// <summary>
            /// Constructor
            /// </summary>
            public ViewModelMainPage()
            {
                // Create Employee List
                pEmployeeList = new List<Employee>();
            }

            /// <summary>
            /// Add Employee
            /// </summary>
            /// <param name="employeeId">Employee Id</param>
            /// <param name="employeeName">Employee Name</param>
            public void AddEmployee(int employeeId, string employeeName)
            {
                // Add Employee
                pEmployeeList.Add(new Employee() { EmployeeId = employeeId, EmployeeName = employeeName });
                // Update Employee List
                OnPropertyChanged(nameof(Employees));
            }
        }

        /// <summary>
        /// Main Page ViewModel
        /// </summary>
        public ViewModelMainPage vmMainPage = null;

        /// <summary>
        /// Constructor
        /// </summary>
        public MainPage()
        {
            InitializeComponent();

            // Create Main Page View Model
            vmMainPage = new ViewModelMainPage();
            // Add Test Employee
            vmMainPage.AddEmployee(1, "Taro Yamada");
            vmMainPage.AddEmployee(2, "Hanako Yamada");
       
            // Bind MainPage View Model
            this.BindingContext = vmMainPage;
        }

        /// <summary>
        /// Add Employee
        /// </summary>
        /// <param name="sender">Event Source</param>
        /// <param name="e">Event Args</param>
        private void btnAdd_Clicked(object sender, EventArgs e)
        {
            // Set Next Employee Id
            int nextId = vmMainPage.Employees.AsEnumerable().Max(x => x.EmployeeId) + 1;
            // Create New Employee
            vmMainPage.AddEmployee(nextId, null);
        
            // Scroll Employee List
            IDispatcherTimer timer;
            timer = Dispatcher.CreateTimer();
            timer.Interval = new TimeSpan(0, 0, 0, 0, 10);
            timer.Tick += (s, e) =>
            {
                timer.Stop();

                MainThread.BeginInvokeOnMainThread(() =>
                {
                    svEmployeeList.ScrollToAsync(slEmployeeList, ScrollToPosition.End, true);
                });
            };
            timer.Start();
        }
    }

[App.xaml.cs]

    MainPage = new NavigationPage(new MainPage());

Below is a video of the verification. (Verification was performed on Android.)

[When using NavigationPage to generate MainPage (.NET MAUI Version)]

https://user-images.githubusercontent.com/125236133/227823981-dc744e88-3ecb-4b7c-9211-01792c30e802.mp4

To reproduce the issue, the Entry should be tapped to focus before pressing the Add Employee button.
After pressing the Add Employee button, the Picker at the top of the screen is 
focused and the Picker Dialog is opened. I don't want this focus shift.
In Xamarin.Forms, such focus movement did not occur.

[When using NavigationPage to generate MainPage (Xamarin.Forms Version)]

https://user-images.githubusercontent.com/125236133/227823630-7f2cddc2-c877-469c-b88c-10eb8b1154b5.mp4

If you change MainPage generation to Shell instead of NavigationPage, this focus movement will not occur.

    MainPage = new AppShell();

[When using Shell to generate MainPage (.NET MAUI Version)]

https://user-images.githubusercontent.com/125236133/227825027-4668a61d-fbd8-4825-a3ee-fe2ddc771694.mp4

There is an obvious difference in focus control between NavigationPage and Shell. 
In Xamarin Forms, NavigationPage was used to generate MainPage, so I migrated to .NET MAUI 
as it is, but I'm having trouble with unintended focus control.

Do you have any good ideas? Thank you.

**Comment by Realgigabyte** [comment]
I think I am having the same issue. Substantially the first entry type control will get focus when the page is set as MainPage. Also the keyboard might show up if the control is a text entry.
